packages/react-dom/src/__tests__/escapeScriptForBrowser-test.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+'use strict';
+
+let React;
+let ReactDOMFizzServer;
+let Stream;
+
+function getTestWritable() {
+  const writable = new Stream.PassThrough();
+  writable.setEncoding('utf8');
+  const output = {result: '', error: undefined};
+  writable.on('data', chunk => {
+    output.result += chunk;
+  });
+  writable.on('error', error => {
+    output.error = error;
+  });
+  const completed = new Promise(resolve => {
+    writable.on('finish', () => {
+      resolve();
+    });
+    writable.on('error', () => {
+      resolve();
+    });
+  });
+  return {writable, completed, output};
+}
+
+describe('escapeScriptForBrowser', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    ReactDOMFizzServer = require('react-dom/server');
+    Stream = require('stream');
+  });
+
+  it('"<[Ss]cript" strings are replaced with unicode escaped lowercase s or S depending on case', () => {
+    const {writable, output} = getTestWritable();
+    const {pipe} = ReactDOMFizzServer.renderToPipeableStream(<div />, {
+      bootstrapScriptContent:
+        '"prescription pre<scription preScription pre<Scription"',
+    });
+    pipe(writable);
+    jest.runAllTimers();
+    expect(output.result).toMatch(
+      '<div></div><script>"prescription pre<\\u0073cription preScription pre<\\u0053cription"</script>',
```

Comment:
We have a real HTML parser in JSDOM. You could write a test that assert that you're getting the expected parsing result out of it instead of hardcoding the expected result.

That will make it easier to expand this testing and have existing tests work if we ever change this.

It'll also better communicate to future us what were trying to prevent here exactly.